track_generator/pytrack_generator/bindings.cpp

```diff
@@ -0,0 +1,122 @@
+#include "routing/route.hpp"
+#include "routing/routing_callbacks.hpp"
+#include "routing/routing_quality/utils.hpp"
+
+#include "platform/platform.hpp"
+
+#include "geometry/latlon.hpp"
+
+#include <exception>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wreorder"
+#pragma GCC diagnostic ignored "-Wunused-local-typedefs"
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-local-typedef"
+#endif
+
+#include "pyhelpers/module_version.hpp"
+#include "pyhelpers/vector_list_conversion.hpp"
+
+#include <boost/python.hpp>
+#include <boost/python/exception_translator.hpp>
+#include <boost/python/suite/indexing/vector_indexing_suite.hpp>
+
+using namespace std;
+
+namespace
+{
+void TranslateRuntimeError(std::runtime_error const & e)
+{
+  PyErr_SetString(PyExc_RuntimeError, e.what());
+}
+
+struct Params
+{
+  Params(string const & data, string const & userResources) : m_dataPath(data), m_userResourcesPath(userResources)
+  {
+    if (m_dataPath.empty())
+      throw runtime_error("Specify -data_path");
+
+    if (m_userResourcesPath.empty())
+      throw runtime_error("Specify -user_resources_path");
+  }
+
+  string DebugPrint() const
+  {
+    ostringstream ss;
+    ss << "Params(data path: " << m_dataPath << ", user resources path: " << m_userResourcesPath << ")";
+    return ss.str();
+  }
+
+  string m_dataPath;
+  string m_userResourcesPath;
+};
+
+using Track = vector<ms::LatLon>;
+
+Track GetTrackFrom(routing::Route const & route)
+{
+  CHECK(route.IsValid(), ());
+  auto const & segments = route.GetRouteSegments();
+  Track res;
+  res.reserve(segments.size());
+  for (auto const & s : segments)
+    res.emplace_back(MercatorBounds::ToLatLon(s.GetJunction().GetPoint()));
+
+  return res;
+}
+
+struct Generator
+{
+  explicit Generator(Params const & params) : m_params(params)
+  {
+    Platform & pl = GetPlatform();
+    pl.SetWritableDirForTests(m_params.m_dataPath);
+    pl.SetResourceDir(m_params.m_userResourcesPath);
+  }
+
+  Track Generate(boost::python::object const & iterable) const
+  {
+    using namespace routing_quality;
+
+    Track const coordinates = python_list_to_std_vector<ms::LatLon>(iterable);
+    auto result = GetRoute(FromLatLon(coordinates), routing::VehicleType::Pedestrian);
+    if (result.m_code != routing::RouterResultCode::NoError)
+      throw runtime_error("Can't find route");
```

Comment:
```suggestion
      throw runtime_error("Could not build route");
```
В routing/routing_quality/utils.hpp всё-таки в комменте термин Build